source/routing/raptor_api.cpp

```diff
@@ -1369,97 +1370,87 @@ void filter_late_journeys(RAPTOR::Journeys& journeys, const NightBusFilter::Para
 //
-// returns `true` if the journey has been modified
-bool shorten_journey(navitia::routing::Journey& journey,
-                     const std::string& stop_area_uri,
-                     const navitia::type::VehicleJourney* vj_to_skip,
-                     const map_stop_point_duration& departures,
-                     const map_stop_point_duration& destinations,
-                     const bool clockwise) {
-    auto res = get_and_update_visited_section(journey, stop_area_uri, vj_to_skip, departures, destinations, clockwise);
-    if (res.first) {
-        if (clockwise) {
-            journey.sections.erase(journey.sections.begin() + res.second + 1, journey.sections.end());
-        } else {
-            journey.sections.erase(journey.sections.begin(), journey.sections.begin() + res.second);
+// returns `true` if the section has been modified
+bool shorten_section_anticlockwise(navitia::routing::Journey::Section& section,
+                                   const std::string& first_stop_area_uri,
+                                   const navitia::type::VehicleJourney* first_vj,
+                                   const map_stop_point_duration& fallbacks) {
+    log4cplus::Logger logger = log4cplus::Logger::getInstance(LOG4CPLUS_TEXT("logger"));
+
+    // because of stay-ins, we may have several vj in one section, we have to scan the stop times of all vjs
+    auto order = type::RankStopTime(section.get_out_st->vehicle_journey->stop_time_list.size() - 1
+                                    - section.get_out_st->order().val);
+    for (const auto* vj = section.get_out_st->vehicle_journey; vj; (vj = vj->prev_vj, order = type::RankStopTime(0))) {
+        if (vj == first_vj) {
+            return false;
+        }
+
+        if (section.get_in_dt < section.get_in_st->boarding_time) {
+            LOG4CPLUS_ERROR(logger, "Error : section.get_in_dt < section.get_in_st->boarding_time");
+            return false;
+        }
+        // determine midnigth of the day at which the vj is used
+        // WARNING : this base_dt may need to be increased in case there is some stay-ins vehicle
+        // with stop_times that are past midnight (not handled yet)
+        navitia::DateTime base_dt = section.get_in_dt - section.get_in_st->boarding_time;
+        for (const auto& st :
+             boost::make_iterator_range(vj->stop_time_list.rbegin() + order.val, vj->stop_time_list.rend())) {
+            // check stop_area is the same and it's possible to use stop_point to get in PT
+            if (st.stop_point->stop_area->uri == first_stop_area_uri
+                && navitia::contains(fallbacks, routing::SpIdx{st.stop_point->idx}) && st.pick_up_allowed()) {
+                section.get_in_st = &st;
+                section.get_in_dt = base_dt + st.boarding_time;
+
+                return true;
+            }
         }
     }
+
+    LOG4CPLUS_ERROR(logger, "Error occurred when modifying backtracking journeys anticlockwise");
```

Comment:
Same as above, this is not an error anymore to reach this line.

Reply:
See response above :wink: